espnet2/svs/singing_tacotron/encoder.py

```diff
@@ -0,0 +1,264 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# Copyright 2019 Nagoya University (Tomoki Hayashi)
+# Copyright 2023 Renmin University of China (Yuning Wu)
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+"""Singing Tacotron encoder related modules."""
+
+import torch
+from torch.nn.utils.rnn import pack_padded_sequence, pad_packed_sequence
+
+
+def encoder_init(m):
+    """Initialize encoder parameters."""
+    if isinstance(m, torch.nn.Conv1d):
+        torch.nn.init.xavier_uniform_(m.weight, torch.nn.init.calculate_gain("relu"))
+
+
+class Encoder(torch.nn.Module):
```

Comment:
Ditto, if possible, let's inherit Tomoki's original one

Reply:
For decoder, the transition token in GDCA makes it different from tacotron's.

Reply:
Tacotron's encoder gets text as input and we use content embedding as input. The content embedding can be different for GDCA and others. Besides, a new duration_encoder is added here.